### DISCUSSION TITLE
Best practices for big S4 objects and targets?

### DISCUSSION THREAD

**Discussion by c1au6i0** [ropensci/targets#887] [Help] [answered]
I am doing some scRNA using the package {Seurat} that implements a specific `S4` class and  I am not really sure how to organize  this in targets. Let me explain.

Depending on the experiment, Seurat objects can be quite big and most of the analysis are done modifying single slots of the object.
Having each target creating a new object in which only one slot is modified  would end up being very memory hungry...
What do you guys do in these scenarios?

**Comment by wlandau** [comment, marked as answer]
That’s a tough one because it breaks the paradigm. {targets} thinks of a data analysis pipeline as a collection of data transformations where the output is immutable and distinct from the input. If you can save smaller or more easily compressible intermediate pieces as your intermediate target outputs and build/tear down the S4 objects in a manner completely contained within each target, that would be ideal. You might also have a look at arguments ‘memory’ and ‘garbage_collection’ in tar_option_set().
